test/test_linearizer.py

```diff
@@ -307,6 +307,48 @@ def test_partial_opt_multireduce(self):
         if i == 0: continue
         assert ranges[i-1] != u, f"multireduce nested the ranges! {ranges[i-1], {u}}"
 
+  @unittest.skipIf(CI and Device.DEFAULT in {"AMD"}, "AMD CI doesn't support multiple sync threads yet")
+  @unittest.skipUnless(Device[Device.DEFAULT].renderer.has_shared, "test requires shared")
+  @unittest.skipUnless(Device[Device.DEFAULT].renderer.has_local, "test requires locals")
+  def test_multireduce_smem_usage(self):
```

Comment:
this test makes sure that:
1) the new smem checker correctly calculates smem usage
2) that it still throws a `KernelOptError` if max_shared is exceeded

Reply:
It is SUPER SLOW because it uses so many upcasts to drive smem usage up to the boundary without pushing the thread group boundaries. I'm not sure if it makes sense to include in tests for that reason but I do think it's at least relevant to test the PR